core/frontend/src/RealityDataSource.ts

```diff
@@ -265,3 +285,87 @@ export class RealityDataSourceProviderRegistry {
     return this._providers.get(name);
   }
 }
+
+/**
+ * Options for creating a Google Photorealistic 3D Tiles (GP3DT) reality data source provider.
+ * @beta
+ */
+export interface RealityDataSourceGP3DTProviderOptions {
+  /** Google Map Tiles API Key used to access GP3DT. */
+  apiKey?: string;
+  /** Function that returns an OAuth token for authenticating with GP3DT. This token is expected to not contain the "Bearer" prefix. */
+  getAuthToken?: () => Promise<string | undefined>;
```

Comment:
Why is the function permitted to return an undefined auth token and what do you do in that situation?